README.md

```diff
@@ -0,0 +1,133 @@
+# ts-routes
+
+![npm](https://img.shields.io/npm/v/ts-routes)
+[![Actions Status](https://github.com/leancodepl/ts-routes/workflows/build/badge.svg)](https://github.com/leancodepl/ts-routes/actions)
+
+Helper library for constructing strongly typed parameterized routing paths. It prevents you from passing hardcoded
+strings with routes across the app.
+
+ts-routes is independent on routing libraries and can be used together with e.g. React Router DOM or Vue Router.
+
+## Installation
+
+```
+npm install ts-routes
+yarn add ts-routes
+```
+
+## Quick start
+
+```js
+import { createRouting, number, query, uuid } from 'ts-routes';
+
+const routes = createRouting({
+    products: segment`/products`,
+    users: segment`/users/${number('userId')}`,
+    items: {
+        ...segment`/items${query({ filter: false })}`,
+        children: {
+            item: segment`/${uuid('itemId')}`,
+        },
+    },
+} as const);
+
+routes.products(); // '/products'
+routes.users({ userId: '10' }) // '/users/10'
+routes.items({}, { filter: 'new' }) // '/items?filter=new'
+routes.items.item({ itemId: '12d66718-e47c-4a2a-ad5b-8897def2f6a7' }) // '/items/12d66718-e47c-4a2a-ad5b-8897def2f6a7'
```

Comment:
I think it would also be good to show off the `.pattern` part of the library, especially combined with more useful features like the guid/number arg patterns and the way it works with e.g. react-router

Reply:
Yep, I completely forgot about that. I'll mention it in the docs and add tests.